### ISSUE TITLE
Since DST, consistently receiving Error in log

### ISSUE THREAD

**Issue by lucashby** [HangfireIO/Hangfire#529]
Since the time change Sunday morning, I keep seeing log entries with a message similar to this:

Error occurred during execution of 'RecurringJobScheduler' process. Execution will be retried (attempt 200 of 2147483647) in 00:05:00 seconds.

The full details:

```
System.ArgumentException: The supplied DateTime represents an invalid time.  For example, when the clock is adjusted forward, any time in the period that is skipped is invalid.
Parameter name: dateTime
   at System.TimeZoneInfo.ConvertTime(DateTime dateTime, TimeZoneInfo sourceTimeZone, TimeZoneInfo destinationTimeZone, TimeZoneInfoOptions flags, CachedData cachedData)
   at Hangfire.Server.ScheduleInstant.<GetNextInstants>b__0(DateTime x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Hangfire.Server.ScheduleInstant.GetNextInstants(Nullable`1 lastInstant)
   at Hangfire.Server.RecurringJobScheduler.TryScheduleJob(JobStorage storage, IStorageConnection connection, String recurringJobId, IReadOnlyDictionary`2 recurringJob)
   at Hangfire.Server.RecurringJobScheduler.Execute(BackgroundProcessContext context)
   at Hangfire.Server.ServerProcessExtensions.Execute(IServerProcess process, BackgroundProcessContext context)
   at Hangfire.Server.AutomaticRetryProcess.Execute(BackgroundProcessContext context)
```

I don't see anywhere in the stack any reference to my recurring job code so I'm not sure where this error is coming from.

Any help is much appreciated.

EDIT:  After some more looking around, it looks like it is not running ANY of my recurring jobs.

**Comment by lruckman** [comment]
We are also experiencing this issue as well. Tried restarting the application pool but no luck.

**Comment by lruckman** [comment]
OK a quick workaround, under "recurring jobs" select all jobs and delete them. Then restart the application pool which causes the Hangfire _startup.cs _file to get called and in our case all of our jobs are registered there so they all get re-added again.

**Comment by lucashby** [comment]
Yes, we found the same workaround.  I am hoping this can be fixed so next time we don't have to work around the issue.

**Comment by odinserj** [comment]
This bug was introduced with the timezone support for recurring jobs. Don't understand how could I have missed this. Does anybody want to fix this and add some unit tests?

**Comment by nadeembs** [comment]
So i did the same workaround deleted all the entries from the Job table. the error stopped but When i restarted the recurring Job issue came again.

Any ideas or workaround to fix this?

**Comment by odinserj** [comment]
NCrontab may return invalid date and time values, since it doesn't know anything about time zones and/or daylight savings. This change allows Hangfire to skip those values when calculating next execution time for recurring jobs.

The fix will be released today in 1.5.4 version.
